### ISSUE TITLE
Readiness probe failed

### ISSUE THREAD

**Issue by KaiWedekind** [codecentric/helm-charts#22]
Hi,

I'm trying to install Keycloak with Postgres and my keycloak pod failed with "Readiness probe failed: Get http://192.168.0.53:8080/auth/realms/master: dial tcp 192.168.0.53:8080: connect: connection refused"

Checking the logs, I'm getting the following:

```
19:11:20,788 INFO  [org.jboss.modules] (CLI command executor) JBoss Modules version 1.8.7.Final
LogManager error of type GENERIC_FAILURE: Failed to move file /opt/jboss/keycloak/standalone/log/server.log to /opt/jboss/keycloak/standalone/log/server.log.2019-03-20.
java.nio.file.FileSystemException: /opt/jboss/keycloak/standalone/log/server.log -> /opt/jboss/keycloak/standalone/log/server.log.2019-03-20: Operation not permitted   

...

19:11:20,929 INFO  [org.jboss.msc] (CLI command executor) JBoss MSC version 1.4.5.Final
19:11:20,950 INFO  [org.jboss.threads] (CLI command executor) JBoss Threads version 2.3.2.Final
19:11:21,402 INFO  [org.jboss.as] (MSC service thread 1-1) WFLYSRV0049: Keycloak 5.0.0 (WildFly Core 7.0.0.Final) starting
19:11:21,568 INFO  [org.jboss.vfs] (MSC service thread 1-2) VFS000002: Failed to clean existing content for temp file provider of type temp. Enable DEBUG level log to find what caused this
19:11:21,656 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-2) MSC000001: Failed to start service jboss.deployment-mount-provider: org.jboss.msc.service.StartException in service jboss.deployment-mount-provider: WFLYSRV0113: Failed to create temp file provider
	at org.jboss.as.server.deployment.DeploymentMountProvider$Factory$ServerDeploymentRepositoryImpl.start(DeploymentMountProvider.java:127)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1738)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1700)
	at org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1558)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1985)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1487)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1364)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: VFS000014: Could not create directory for root '/opt/jboss/keycloak/standalone/tmp/vfs/temp' (prefix 'temp', suffix '') after 10 attempts
	at org.jboss.vfs.TempFileProvider.createTempDir(TempFileProvider.java:157)
	at org.jboss.vfs.TempFileProvider.create(TempFileProvider.java:109)
	at org.jboss.as.server.deployment.DeploymentMountProvider$Factory$ServerDeploymentRepositoryImpl.start(DeploymentMountProvider.java:125)
	... 8 more
```
I'd love to know what's wrong with my setup.

Thank you !

**Comment by BookAdvisorIO** [comment]
What solved my problem was setting fsGroup to zero in securityContext:

```
securityContext:
    runAsUser: 1000
    fsGroup: 0
    runAsNonRoot: true
```

Keycloak was not able to create folders with fsGroup: 1000 (jboss), it had to be 0 (root).

```
drwxrwxr-x 11 jboss root  4096 May 12 00:21 keycloak
```

**Comment by KaiWedekind** [comment]
What solved my problem was setting fsGroup to zero in securityContext:

```
securityContext:
    runAsUser: 1000
    fsGroup: 0
    runAsNonRoot: true
```

Keycloak was not able to create folders with fsGroup: 1000 (jboss), it had to be 0 (root).

```
drwxrwxr-x 11 jboss root  4096 May 12 00:21 keycloak
```

**Comment by unguiculus** [comment]
Can you share some more details on your environment? Do you use Istio by any chance? See #24.

**Comment by RAbraham** [comment]
I'm having the same issue. @KaiWedekind 's solution unfortunately didn't work in my context. I'm testing it locally on MacOS. The only change I made was to point to an existing postgres local instance. Is this issue caused if the database `keycloak` is an empty database? Does the keycloak deployment try to populate an empty database with keycloak specific schemas?

**Comment by KaiWedekind** [comment]
I use Istio, indeed. My issue wasn't initializing iptables though, my keycloak container just **Could not create directory for root '/opt/jboss/keycloak/standalone/tmp/vfs/temp'** as included in the log snapshot.

**Comment by unguiculus** [comment]
Should be fixed by #25.
